File: go/internal/domains/health/health.go
// Package health — the liveness probe.
package health

import (
	"net/http"

	"app/internal/core"
)

func HealthCheck(w http.ResponseWriter, r *http.Request) {
	// read-scope: public — liveness probe, returns only {status}, no domain state.
	core.WriteJSON(w, 200, map[string]string{"status": "ok"})
}

File: python/app_pkg/domains/health.py
from fastapi import APIRouter
from pydantic import BaseModel

router = APIRouter(tags=["health"])


class HealthResponse(BaseModel):
    status: str


@router.get("/health", response_model=HealthResponse)
def health() -> HealthResponse:
    # read-scope: public — liveness probe, returns only {status}, no domain state.
    return HealthResponse(status="ok")
